src/string_util.h

```diff
@@ -32,6 +32,100 @@ inline std::string StrCat(Args&&... args) {
   return ss.str();
 }
 
+#if defined(__ANDROID__) && defined(__GLIBCXX__)
```

Comment:
can you define a BENCHMARK_OS_ANDROID where we define the other OS constants (src/internal_macros.h)?

i'm not sure we currently differentiate on libcxx but maybe we should add something there too.